registry/handlers/app.go

```diff
@@ -102,6 +108,10 @@ func NewApp(ctx context.Context, config *configuration.Configuration) *App {
 		isCache: config.Proxy.RemoteURL != "",
 	}
 
+	if err := app.traceHTTPMiddleware(); err != nil {
+		panic(err)
+	}
```

Comment:
I assume we need to `panic()` here because `NewApp()` doesn't have an error return; perhaps we should consider changing its signature so that we can return the error instead.